src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Methodologies/MethodologyApprovalService.cs

```diff
@@ -99,11 +99,14 @@ public class MethodologyApprovalService : IMethodologyApprovalService
 
                         methodologyVersion.Updated = DateTime.UtcNow;
 
-                        var isPubliclyAccessible = await _methodologyVersionRepository.IsPubliclyAccessible(methodologyVersion.Id);
+                        // We cannot rely on Methodology.LatestPublishedVersionId, as it may now be incorrect,
+                        // if we are approving and publishing this methodology version.
+                        var isToBePublished = await _methodologyVersionRepository.IsPubliclyAccessible(methodologyVersion);
```

Comment:
I had exactly the same thought myself about renaming it, and like you struggled to find something I preferred.

I renamed the variable `isToBePublished` and not the method because I thought `IsPubliclyAccessible` might still be more accurate if someone uses the method for the explicit purpose of checking whether it's accessible. But thinking on it further, now that we have `LatestPublishedVersion`, I cannot think why anyone would use `IsPubliclyAccessible` except for when the `LatestPublishedVersion` has changed. And that would only happen if a new `MethodologyVersion` is being published.

So I've renamed as you suggest. 

My only concern about moving it and making it private is the extra time it'll take to adapt the unit tests. The scenarios covered by the existing IsToBePublished tests would need to be adapted for `MethodologyApprovalService#UpdateStatus` so we still cover the same scenarios. But I'll leave comments to move and make it private and assume that's what we're doing for now. If adapting the unit tests becomes a major headache, I'll complain when I come to do it :)

Reply:
OK, sounds good to me!